src/main/java/org/alliancegenome/curation_api/services/base/BaseAssociationDTOCrudService.java

```diff
@@ -1,17 +1,12 @@
 package org.alliancegenome.curation_api.services.base;
 
 import org.alliancegenome.curation_api.dao.base.BaseEntityDAO;
-import org.alliancegenome.curation_api.enums.BackendBulkDataProvider;
-import org.alliancegenome.curation_api.exceptions.ObjectUpdateException;
+import org.alliancegenome.curation_api.interfaces.crud.BaseUpsertServiceInterface;
 import org.alliancegenome.curation_api.model.entities.base.AuditedObject;
 import org.alliancegenome.curation_api.model.ingest.dto.base.BaseDTO;
 
-public abstract class BaseAssociationDTOCrudService<E extends AuditedObject, T extends BaseDTO, D extends BaseEntityDAO<E>> extends BaseEntityCrudService<E, D> {
+public abstract class BaseAssociationDTOCrudService<E extends AuditedObject, T extends BaseDTO, D extends BaseEntityDAO<E>> extends BaseEntityCrudService<E, D> implements BaseUpsertServiceInterface<E, T>{
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.GenericWhitespaceCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'>' is followed by an illegal character.